### ISSUE TITLE
Steps before orientation assessment

### ISSUE THREAD

**Issue by VasilevIvanVladimirovich** [PBernalPolo/test_MKF#4]
Good afternoon.
I really liked your work posted on YouTube, thank you very much.

At the moment I am studying the topic of IMU and orientation search, and I have a question, what did you use to achieve such accuracy (very small drift) in a YouTube video ?
I saw a demonstration of your temperature calibration on YouTube, and I was wondering what steps you took before demonstrating Kalman filters.
Сonnecting the sensor -> ?? -> temperature calibration -> ?? ->orientation estimation using kalman filter

Could you tell us about it ?

**Comment by pbernalpolo** [comment]
Hi @VasilevIvanVladimirovich ,

I am glad you liked the YouTube videos, and I hope you find them insightful. :slightly_smiling_face: 

Regarding your question about the procedure I followed, I'm not sure what you really want to know.

- If you are interested in the sequence of operations that the measurements are subjected to, it looks like this:

1. Calibrate the IMU. This is done offline. This can range from taking the default calibration presented in the datasheet, to a complex calibration model that includes biases, scale factor corrections, cross-axis sensitivities, and even temperature dependence of the previously mentioned corrections. Results will be better if calibration includes temperature dependence, but this requires some special hardware. The next points happen online.
2. Take a sensor measurement in raw format.
3. Use the IMU calibration to obtain the calibrated measurement.
4. Perform the prediction step of the Kalman filter.
5. Use your measurements to update the state of the Kalman filter.

- If you are asking about the learning path, then this was a lot more chaotic. I remember starting to just connect the sensors and obtain some measurements. I got a little familiar with the errors of the IMUs. For example, I left the IMU rest on top of the table, and I measured the angular velocity to find out that there was an offset. I also measured the magnitude of the acceleration and found out that the module of the acceleration was dependent on the orientation of the sensor (this would not be the case for an ideal sensor). I started to think and research about calibration methods. At this time I had only thought about offline calibration methods. This is a topic that I left aside until almost the final part of my PhD. I started learning about Kalman filtering, and how to apply it to orientation estimation. This took some time. After the first good paper (https://www.mdpi.com/1424-8220/19/1/149) I tried to do transform the IMU accelerations to an inertial reference frame and do dead reckoning. However, I didn't managed to do it, and I thought that maybe, resuming that topic I left aside things could be improved. This way I continued researching about calibration algorithms. The result was a second good paper (https://ieeexplore.ieee.org/abstract/document/8861358). I finished my PhD at that point, and today I am still learning... :smiley: 

I don't know if any of these responses answer your question. I hope yes :grimacing:
